src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -300,7 +302,13 @@ private[snowflake] class JDBCWrapper {
       case java.sql.Types.CLOB => StringType
       case java.sql.Types.DATALINK => null
       case java.sql.Types.DATE => DateType
-      case java.sql.Types.DECIMAL if precision != 0 || scale != 0 => DecimalType(precision, scale)
+      case java.sql.Types.DECIMAL if precision != 0 || scale != 0 => {
+        if (precision > DecimalType.MAX_PRECISION) {
+          DecimalType(DecimalType.MAX_PRECISION, scale + (precision - DecimalType.MAX_SCALE))
```

Comment:
Not sure why we'd increase scale here. I understand we might limit precision, but doesn't it mean scale should still stay the same?